controllers/core/openstackclient_controller.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	keystonev1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/configmap"
+	env "github.com/openstack-k8s-operators/lib-common/modules/common/env"
+	helper "github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	labels "github.com/openstack-k8s-operators/lib-common/modules/common/labels"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/secret"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	"github.com/openstack-k8s-operators/openstack-operator/pkg/openstack"
+)
+
+// GetClient -
+func (r *OpenStackClientReconciler) GetClient() client.Client {
+	return r.Client
+}
+
+// GetKClient -
+func (r *OpenStackClientReconciler) GetKClient() kubernetes.Interface {
+	return r.Kclient
+}
+
+// GetLogger -
+func (r *OpenStackClientReconciler) GetLogger() logr.Logger {
+	return r.Log
+}
+
+// GetScheme -
+func (r *OpenStackClientReconciler) GetScheme() *runtime.Scheme {
+	return r.Scheme
+}
+
+// OpenStackClientReconciler reconciles a OpenStackClient object
+type OpenStackClientReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	Kclient kubernetes.Interface
+	Log     logr.Logger
+}
+
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core.openstack.org,resources=openstackclients/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete;
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete;
```

Comment:
for now could be just `get;list;watch;` ? don't seem to `create;patch;delete`

Reply:
fixed in the latest PR